bigframes/session/__init__.py

```diff
@@ -1154,35 +1153,14 @@ def _read_pandas_load_job(
         )
         self._start_generic_job(load_job)
 
-        ordering = order.ExpressionOrdering(
-            ordering_value_columns=tuple([order.ascending_over(ordering_col)]),
-            total_ordering_columns=frozenset([ordering_col]),
-            integer_encoding=IntegerEncoding(True, is_sequential=True),
-        )
-        table_expression = self.ibis_client.table(  # type: ignore
-            load_table_destination.table_id,
-            schema=load_table_destination.dataset_id,
-            database=load_table_destination.project,
-        )
-
-        # b/297590178 Potentially a bug in bqclient.load_table_from_dataframe(), that only when the DF is empty, the index columns disappear in table_expression.
-        if any(
-            [new_idx_id not in table_expression.columns for new_idx_id in new_idx_ids]
-        ):
-            new_idx_ids, idx_labels = [], []
-
-        column_values = [
-            table_expression[col]
-            for col in table_expression.columns
-            if col != ordering_col
-        ]
-        array_value = core.ArrayValue.from_ibis(
-            self,
-            table_expression,
-            columns=column_values,
-            hidden_ordering_columns=[table_expression[ordering_col]],
-            ordering=ordering,
-        )
+        destination_table = self.bqclient.get_table(load_table_destination)
+        array_value = core.ArrayValue.from_table(
+            table=destination_table,
+            # TODO: Generate this directly from original pandas df.
```

Comment:
Would we actually want to do that, though? Presumably the type in BQ is the source of truth for us.

Reply:
I was thinking about if we virtualize some datatypes such as microsecond intervals, which would be represented by integer in bq